### DISCUSSION TITLE
importing 100k existing photos: questions and practical concerns

### DISCUSSION THREAD

**Discussion by jeppo7712** [immich-app/immich#3077] [General]
Immich looks very promising and I want to give it a try after having used Photoprism for some time. I have an existing photo library, structured in folders named after events or type of photos inside. There are about 100k files inside: JPEGs, HEIC files with corresponding MOV file, RAW files, etc.

I have used the "Read-only Gallery [Experimental]" method to do the import. Next to that I have enabled photo backup in the iOS app. 

I have several questions and concerns:

-The import itself was rather quick, but the processing of photos is already taking many days. I don't know when it's going to be finished to be honest. Is this normal?
-While the processing is ongoing, the iOS app is behaving strange. Not all photos are shown, not all videos are working, syncing with the server takes some time, sometimes it just tells me it's the first time I use the app when I open it while it's not, etc... I assume this is normal if not all photos are processed?
-The database size is already huge. I don't know how big it's going to be in the end but it's already bigger than the Photoprism db.
-I assume I can say goodbye to my folder structure? I.e. Am I correct to assume Immich will just show a flat structure?
-Most of my iPhone photos were already backed up before I started with Immich. They're part of the library I'm importing. I want to use Immich for future backups, but I couldn't find a way in the app to only back up newly taken photos. So the app is now attempting to backup all my iPhone photos again... This is not only unnecessary, but it's also not really succeeding because Immich seems to be too busy processing the database import. Is there a way to tell the app to backup only new photos?

**Comment by alextran1502** [comment]
- You can check the status of the processing jobs in Administration > Jobs panels on the web.

- The speed of the processing speed depends on your CPU power as server will generate thumbnail files, encodes videos and processing machine learning jobs. All of those jobs are being done by the CPUs at the moment

- We have client side hashing put in place for the mobile app but haven't put in a mechanism to use those info to identify which identical files are on the server before uploading yet. Once this is implemented, it will be able to only upload the photos that are not on the server.

- Database might be bigger be cause we store more metadata to make things work nicely. However this should be negligible compare to the size of your gallery.

So, we have plan to optimize all of the above-mentioned points in the future, such as leveraging GPU for CPU intensive tasks and make loading on the mobile app faster 

Cheers!

😀

**Reply by jeppo7712** [reply]
Thanks Alex. I'll keep you posted on how it goes. Also, having a folder view in a future update would great! As you can imagine, 100k photos in a flat structure is unworkable (re-assigning them to albums after import would be too much work).

**Reply by etnoy** [reply]
Hey jeppo. I'm in a similar situation to you, and I'm a dev on this project. I have some ideas for making this better but it's not coming anythime soon. Perhaps I can bounce some ideas with you in the future?

**Reply by jeppo7712** [reply]
Hi etnoy, sure I would be happy to help!
BTW, In the meantime the processing of my imported library failed. I'm still missing many transcoded videos and thumbnails, even though I pressed "missing" several times in the "jobs" tab. The counter goes back to 0. Also the backup of iOS photos has never completed. If I find the time I can share the log

**Comment by justin13888** [comment]
I've also been using PhotoPrism for a while and got terabytes of HEIC and JPG photos ans well has HEVC and Sony ARW (RAW) videos.

Immich's native mobile app features intrigue me and so I'd like to give it a try.

Question: Is there anything I should know before importing my whole library? Is it typically CPU, GPU, disk bound when importing the media (maybe it needs to index?). I have a lot of CPU resources to use to import though. Final deployment for image is on a Kubernetes cluster with data attached to a centralized NFS share (is that a reasonable idea)?

**Comment by justin13888** [comment]
Additionally, as the user, are there any safety measures to prevent images from being accidentally wiped? Also, is there any thing I should know to recreate all my albums, libraries, etc.?

**Comment by aisbergde** [comment]
@justin13888 - just mount your existing collection as external libraries and test immich. Because they are read only, you can't lost data.
